ui/app/mainui/Popups.qml

```diff
@@ -1,28 +1,127 @@
-import QtQuick 2.14
+import QtQuick 2.15
 
 import AppLayouts.Chat.popups 1.0
+import AppLayouts.Profile.popups 1.0
+
 import shared.popups 1.0
+import shared.status 1.0
 
 import utils 1.0
 
 QtObject {
     id: root
 
-    /* required */ property var rootStore
+    required property var popupParent
+    required property var rootStore
+
+    property var activePopupComponents: []
+
+    Component.onCompleted: {
+        Global.openSendIDRequestPopup.connect(openSendIDRequestPopup)
+        Global.openOutgoingIDRequestPopup.connect(openOutgoingIDRequestPopup)
+        Global.openIncomingIDRequestPopup.connect(openIncomingIDRequestPopup)
+        Global.openInviteFriendsToCommunityPopup.connect(openInviteFriendsToCommunityPopup)
+        Global.openContactRequestPopup.connect(openContactRequestPopup)
+        Global.openChooseBrowserPopup.connect(openChooseBrowserPopup)
+        Global.openDownloadModalRequested.connect(openDownloadModal)
+        Global.openImagePopup.connect(openImagePopup)
+        Global.openProfilePopupRequested.connect(openProfilePopup)
+        Global.openNicknamePopupRequested.connect(openNicknamePopup)
+        Global.blockContactRequested.connect(openBlockContactPopup)
+        Global.unblockContactRequested.connect(openUnblockContactPopup)
+        Global.openChangeProfilePicPopup.connect(openChangeProfilePicPopup)
+        Global.openBackUpSeedPopup.connect(openBackUpSeedPopup)
+        Global.openEditDisplayNamePopup.connect(openEditDisplayNamePopup)
+        Global.openPinnedMessagesPopupRequested.connect(openPinnedMessagesPopup)
+        Global.openCommunityProfilePopupRequested.connect(openCommunityProfilePopup)
+        Global.openPopupRequested.connect(openPopup)
+    }
+
+    function openPopup(popupComponent, params = {}, cb = null) {
+        if (activePopupComponents.includes(popupComponent)) {
```

Comment:
I guess there is no use case for two popups at the same time, but with different params, right?

Reply:
Nope, there must be only one popup of a kind at any time